Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/TestRunner/Native/Job/TestImpactNativeShardedTestJobInfoGenerator.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <TestImpactFramework/TestImpactTestSequence.h>
+
+#include <TestRunner/Native/Job/TestImpactNativeShardedTestJobInfoGenerator.h>
+
+namespace TestImpact
+{
+    NativeShardedInstrumentedTestRunJobInfoGenerator::NativeShardedInstrumentedTestRunJobInfoGenerator(
+        const JobInfoGenerator& jobInfoGenerator,
+        size_t maxConcurrency,
+        const RepoPath& sourceDir,
+        const RepoPath& targetBinaryDir,
+        const NativeShardedArtifactDir& artifactDir,
+        const RepoPath& testRunnerBinary,
+        const RepoPath& instrumentBinary,
+        CoverageLevel coverageLevel)
+        : NativeShardedTestRunJobInfoGeneratorBase<NativeInstrumentedTestRunner>(
+            jobInfoGenerator,
+            maxConcurrency,
+            sourceDir,
+            targetBinaryDir,
+            artifactDir,
+            testRunnerBinary)
+        , m_instrumentBinary(instrumentBinary)
+        , m_coverageLevel(coverageLevel)
+    {
+    }
+
+    RepoPath NativeShardedInstrumentedTestRunJobInfoGenerator::GenerateShardedTargetCoverageArtifactFilePath(
+        const NativeTestTarget* testTarget, size_t shardNumber) const
+    {
+        auto artifactFilePath = GenerateTargetCoverageArtifactFilePath(testTarget, m_artifactDir.m_shardedCoverageArtifactDirectory);
+        return artifactFilePath.ReplaceExtension(
+            AZStd::string::format("%zu%s", shardNumber, artifactFilePath.Extension().String().c_str()).c_str());
```

Comment:
A DynamicMemory allocations can be avoided here by using the `AZ_STRING_ARG` instead of `String()` with `%.*s` argument.
Also the final `.c_str()` isn't needed. An AZStd::string_view can be made directly
```suggestion
        return artifactFilePath.ReplaceExtension(
            AZStd::string_view(AZStd::string::format("%zu%.*s", shardNumber, AZ_STRING_ARG(artifactFilePath.Extension().Native()).Native()));
```

Reply:
Fixed (with fixing of syntax) 👍